tests/test_agent_manager.py

```diff
@@ -313,7 +322,28 @@ async def test_api(init_dataclasses_and_load_schema):
     assert_equal_ish(shouldbe, all_agents, sortby=["hostname", "name"])
     agentid = all_agents["processes"][0]["sid"]
 
-    code, all_agents = await am.list_agent_processes(env.id, None)
+    start = agentid
+    code, all_agents_processes = await am.list_agent_processes(environment=None, expired=False, start=start)
+    assert code == 200
+    for agent_process in all_agents_processes["processes"]:
+        assert agent_process["expired"] is None
+
+    code, all_agents_processes = await am.list_agent_processes(environment=None, expired=True, start=start)
```

Comment:
There are no expired agentprocesses, so this test will always succeed. Each of these tests should also add an assertion on the number of records that are expected.

Reply:
Do you have any advice on how to do that? From what I understood, an agent should be stopped and a session should expire for this, but I don't know how to do either of those.

Reply:
The `am._register_session` call is already used to register an new session for a certain agent. The `_expire_session` call can be used to expire the session again.

Reply:
That works, thank you! Did it in https://github.com/inmanta/inmanta/pull/2521/commits/e86aca10c0de62a64242c06139f9567d52c02197